opcodes/riscv-dis.c

```diff
@@ -484,12 +485,30 @@ riscv_disassemble_insn (bfd_vma memaddr, insn_t word, disassemble_info *info)
 	  Elf_Internal_Ehdr *ehdr = elf_elfheader (info->section->owner);
 	  xlen = ehdr->e_ident[EI_CLASS] == ELFCLASS64 ? 64 : 32;
 	}
-
+      /*
+       * Set the value of capmode based on the flags that GDB passed or fall
+       * back to the ELF header if not available.
+       */
+      if (info->flags & RISCV_OPCODE_FLAG_CAPMODE_ENABLED)
+	in_capmode = 1;
+      else if (info->flags & RISCV_OPCODE_FLAG_CAPMODE_DISABLED)
+	in_capmode = 0;
```

Comment:
Seems like github's web UI can't handle the (rather odd) GDB indentation rules. I just copied the xlen indentation from above.